plugins/statistics.koplugin/main.lua

```diff
@@ -2973,7 +2974,24 @@ function ReaderStatistics:getCurrentBookReadPages()
     return read_pages
 end
 
-function ReaderStatistics.onSync(local_path, cached_path, income_path)
+function ReaderStatistics:canSync()
+    return self.settings.sync_server ~= nil and self.settings.is_enabled
+end
+
+function ReaderStatistics:onSyncBookStats()
+    if not self:canSync() then return end
+
+    UIManager:show(InfoMessage:new {
+        text = _("Syncing book statistics, this might take a while…"),
+        timeout = 1,
+    })
+
+    UIManager:nextTick(function()
+        SyncService.sync(self.settings.sync_server, db_location, self.onSyncServiceSync)
+    end)
+end
+
+function ReaderStatistics.onSyncServiceSync(local_path, cached_path, income_path)
```

Comment:
I would let this one with its original name `ReaderStatistics.onSync(...)` just because we don't know if there's not anywhere else something throwing a "Sync" event and expecting handlers to be named onSync (I think we don't, fortunately :)
I see `VocabularyBuilder.onSync()` has one with the same name.
I don't think we have any such "sync everything" use (and I don't know if we would ever need/want it - one could make a gesture/profile to trigger the ones he needs).
But may be they deserve a more complex name than just "Sync" ? Or just not have a "on" prefix, so we don't risk triggering them with a `broadcastEvent(Event:new("Sync"))`.

Pinging @weijiuqiao who implemented both these onSync for thoughts.

Anyway, keeping the original name would have avoided us to wonder about all that :) It's not too late :)

Reply:
Right, I will revert that :)

Reply:
Wanted to maintain distinction between dispatcher ecent and sync method